### PR TITLE
ensure logs with follow=1 sends the HTTP response immediately

### LINKED ISSUE #12704: logs with follow=1 should always be replied immediately
When calling `GET /containers/(id)/logs?follow=1` on a container which has **not yet produced any output** the current behaviour of the engine is to delay the HTTP response until the container writes some output.

The client cannot guess if the command is successful until some data is received.

This is problematic for example when you need to make the request in the foreground thread and process its output in a background thread. In python you would write something like:
```py
    gen = client.logs(container, follow=True)
    executor.submit(func, gen)
```
client.logs() returns a generator that is sent to another thread. However if the container log is empty then client.logs() hangs and will return the generator only when some output is available.

### PR DESCRIPTION
**Pull request by a-ba**
closes #12704

### CODE REVIEW

**Comment by jfrazelle** [comment]
can confirm test fails on master but idk about the timeout, maybe a go routine instead idk

File: daemon/logs.go [hunk 1]
```diff
@@ -129,6 +129,11 @@ func (daemon *Daemon) ContainerLogs(name string, config *ContainerLogsConfig) er
 		errors := make(chan error, 2)
 		wg := sync.WaitGroup{}
 
+		// write an empty chunk of data (this is to ensure that the
+		// HTTP Response is sent immediatly, even if the container has
+		// not yet produced any data)
+		io.WriteString(outStream, "")
```
Comment (LK4D4) [inline daemon/logs.go]: This can be just `outStream.Write(nil)`

**Comment by LK4D4** [comment]
I'm not happy with bringing http logic to daemon methods, but I don't see any not-awful method to do it in `api/server/server.go`

File: daemon/logs.go (hunk 1 above)
Comment (LK4D4) [inline daemon/logs.go]: @a-ba Do you interested in fixing my comment?

**Comment by icecrime** [comment]
Carried in #13033, thanks @a-ba!

**Pull request closed without merging**